src/common/tests_dmg_helpers.c

```diff
@@ -130,11 +131,19 @@ daos_dmg_json_pipe(const char *dmg_cmd, const char *dmg_config_file,
 	FILE			*fp = NULL;
 	int			pc_rc, rc = 0;
 
+	/* allow env to set only if not supplied */
 	if (dmg_config_file == NULL)
-		D_ASPRINTF(cmd_base, "dmg -j -i %s ", dmg_cmd);
+		cfg_file = getenv("DMG_CONFIG_FILE");
 	else
-		D_ASPRINTF(cmd_base, "dmg -j -o %s %s ",
-			   dmg_config_file, dmg_cmd);
+		D_STRNDUP(cfg_file, dmg_config_file, MAXPATHLEN);
+
+	if (cfg_file == NULL)
```

Comment:
What if the strndup fails?  This will be NULL but it's really a -DER_NOMEM situation rather than one where we should continue

Reply:
Oops, good catch. I was checking it in a previous iteration but then rearranged things and forgot to re-add it.